src/Mod/Part/App/Geometry.cpp

```diff
@@ -6244,6 +6244,419 @@ GeomArcOfCircle *createFilletGeometry(const GeomLineSegment *lineSeg1, const Geo
+            if (tmpdist < dist) {
+                dist = tmpdist;
+                interpoints = std::make_pair(tmpp1, tmpp2);
+                pos1 = 2;
+                pos2 = 2;
+            }
+        }
+
+        if (dist == INFINITY) {
+            // no coincident was found, try basis curve intersection if GeomTrimmedCurve
+            if (geo1->isDerivedFrom<GeomTrimmedCurve>() && geo2->isDerivedFrom<GeomTrimmedCurve>()) {
+                auto* tcurve1 =static_cast<const GeomTrimmedCurve*>(geo1);
+                auto* tcurve2 = static_cast<const GeomTrimmedCurve*>(geo2);
+
+                try {
+                    if (!tcurve1->intersectBasisCurves(tcurve2, points)) {
+                        return nullptr;
+                    }
+                }
+                catch (Base::CADKernelError& e) {
+                    e.ReportException();
+                    THROWMT(Base::CADKernelError,
+                            QT_TRANSLATE_NOOP("Exceptions",
+                                              "Unable to guess intersection of curves. Try adding "
+                                              "a coincident constraint between the vertices of the "
+                                              "curves you are intending to fillet."))
+                }
+
+                int res = selectintersection(points, interpoints, refPnt1, refPnt2);
+
+                if (res != 0) {
+                    return nullptr;
+                }
+            }
+            else {
+                return nullptr;// not a GeomTrimmedCurve and no coincident point.
+            }
+        }
+
+        // Now that we know where the curves intersect, get the parameters in the curves of those
+        // points
+        double intparam1;
+        double intparam2;
+
+        try {
+            if (!curve1->closestParameter(interpoints.first, intparam1)) {
+                return nullptr;
+            }
+        }
+        catch (Base::CADKernelError& e) {
+            e.ReportException();
+            THROWM(Base::CADKernelError,
+                   "Unable to determine the parameter of the first selected curve at the "
+                   "intersection of the curves.")
+        }
+
+        try {
+            if (!curve2->closestParameter(interpoints.second, intparam2)) {
+                return nullptr;
+            }
+        }
+        catch (Base::CADKernelError& e) {
+            e.ReportException();
+            THROWM(Base::CADKernelError,
+                   "Unable to determine the parameter of the second selected curve at the "
+                   "intersection of the curves.")
+        }
+
+        // get the starting parameters of each curve
+        double spc1 = curve1->getFirstParameter();
+        double spc2 = curve2->getFirstParameter();
+
+        // get a fillet radius if zero was given
+        Base::Vector3d ref21 = refPnt2 - refPnt1;
+
+        if (radius == .0f) {

```

Comment:
Presumably (but correct me if I'm wrong) this means the radius is not set. We can make radius a `std::optional<float>` to make intentions clear.